From 7d671e66cc409f458e72d4668e248c091337ea29 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Mon, 11 Feb 2019 11:52:50 -0500
Subject: [PATCH 132/202] e2freefrag: fix memory leak in scan_online()

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---
 misc/e2freefrag.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/misc/e2freefrag.c b/misc/e2freefrag.c
index 20b79b3..b80b742 100644
--- a/misc/e2freefrag.c
+++ b/misc/e2freefrag.c
@@ -235,7 +235,7 @@ static int scan_online(ext2_filsys fs, struct chunk_info *info,
 			break;
 		fsmap_advance(fsmap);
 	}
-
+	free(fsmap);
 	return 1;
 }
 #else
-- 
2.7.4

